Refactoring + Error Dialogue fix

foreverwoods opened the pull request:
Changes:
- Refactored Files
- Implemented "Error" Dialogue (fail screen) for an empty sorted list (no restaurants match the search selections)
- separated sorting class into multiple classes

TODO:
- documentation
- **FIXED** sorting

prajogt commented on src/main/java/filtering_use_case/interactors/RestaurantFilterFail.java:
```diff
@@ -0,0 +1,8 @@
+package filtering_use_case.interactors;
+
+public class RestaurantFilterFail extends RuntimeException{
+
```
Minor detail, but for consistency Pete and I did -Failed instead of just Fail

prajogt commented on src/main/java/filtering_use_case/interactors/sortAvgStars.java:
```diff
@@ -0,0 +1,22 @@
+                    Comparator.reverseOrder()));
+        }
+    }
+}
```
Rename class to Pascal case

prajogt commented on src/main/java/filtering_use_case/interactors/sortChoices.java:
```diff
@@ -0,0 +1,75 @@
+
+    }
+
+}
```
Pascal Class name

prajogt replied:
Also rename to SortChoicesInteractor to avoid confusion over others, and this class shouldn't extend Sorting, it just takes in a class of type Sorting

prajogt commented on src/main/java/filtering_use_case/interactors/sortName.java:
```diff
@@ -0,0 +1,21 @@
+            Collections.sort(sortedRestaurants, Comparator.comparing(Restaurant::getName, Comparator.reverseOrder()));
+        }
+    }
+}
```
Pascal

prajogt commented on src/main/java/filtering_use_case/interactors/sortPrice.java:
```diff
@@ -0,0 +1,22 @@
+                    Comparator.reverseOrder()));
+        }
+    }
+}
```
Pascal

prajogt reviewed (dismissed):
Sorting methods should be refactored to Pascal naming convention, and some confusion about the actual sort interactor.

foreverwoods commented:
EDIT: Second and Third Commit

Refactored class names to be in PascalCase.
Changed Sorting Implementation to use radiobutton selection to sort
Removed "extends Sorting" from SortChoicesInteractor
Sorting except Ratings (no reviews) was tested to be working

Added code in @henryklinck 's LoginSubmitActionListener to call a HomeScreenView while passing on user and it connects.

foreverwoods commented:
EDIT:

Sorting works
